spec.bs

```diff
@@ -8349,6 +8628,97 @@ scoring signals requests=] |entriesToBatch|:
+      [=list=] of [=strings=].
+  1. Set |partition|["id"] to |partitionId|.
+  1. Let |namespace| be an empty [=map=], whose [=map/keys=] are [=strings=] and [=map/values=] are [=strings=] or
+      [=list=] of [=strings=].
+  1. Set |namespace|["render_url"] to [|request|'s [=trusted scoring signals request/render URL=]].
+  1. Set |namespace|["ad_component_render_urls"] to |request|'s [=trusted scoring signals request/ad
+    component URLs=].
+  1. Set |partition|["namespace"] to |namespace|.
+  1. Let |partitionMetadata| be an empty [=map=], whose [=map/keys=] and [=map/values=] are [=strings=].
+  1. Set |partitionMetadata|["experiment_group_id"] to |firstRequest|'s [=trusted scoring signals
+      request/seller experiment group id=].
+  1. Set |partition|["metadata"] to |partitionMetadata|.
+  1. Set |compressionGroupMap|[|compressionGroupId|][|partitionId|] to |partition|.
+1. [=map/For each=] |id| → |group| in |compressionGroupMap|:
+  1. Let |compressionGroup| be an empty [=map=], whose [=map/keys=] are [=strings=] and [=map/values=] are integers or
+    [=lists=] of [=maps=].
+  1. Set |compressionGroup|["compression_group_id"] to |id|.
+  1. Set |compressionGroup|["partitions"] to an empty [=list=].
+  1. [=list/For each=] |partition| in |group|'s [=map/values=]:
+    1. [=list/Append=] |partition| to |compressionGroup|["partitions"].
+  1. [=list/Append=] |compressionGroup| to |compressionGroups|.
+1. Let |metadata| be an empty [=map=], whose [=map/keys=] and [=map/values=] are [=strings=].
+1. Let |hostname| be the result of [=string/UTF-8 percent-encoding=] the
+  [=serialization of an origin|serialized=] |firstRequest|'s [=trusted scoring signals request/top
+    level origin=] using [=component percent-encode set=].
+1. Set |metadata|["hostname"] to |hostname|.
+1. Let |keyInfo| be the result of [=looking up the server encryption key=] with |firstRequest|'s
+  [=trusted scoring signals request/seller=] and |firstRequest|'s [=trusted scoring signals
+  request/signal coordinator=].
+1. Let (|requestBlob|, |context|) be the result of generating request |keyInfo|, |metadata| and
+  |compressionGroups|. The generation method may follow that described in
+  [Section 2.2.4 of the Protected Audience Key Value Services](https://privacysandbox.github.io/draft-ietf-protected-audience-key-value-service/draft-ietf-protected-audience-key-value-services.html#name-generating-a-request).
+1. Return |requestBlob|, |renderUrlIdMap| and |context|.
+
+</div>
+
+<div algorithm>
+
+To <dfn>build batched trusted key value scoring signals request body</dfn> given a non-empty [=list=] of
+[=trusted scoring signals requests=] |entriesToBatch|:
+
+1. Let |firstEntry| be |entriesToBatch|[0].
+1. Let |renderURLs| be an empty [=ordered set=] of [=strings=].
+1. Let |adComponentRenderURLs| be an empty [=ordered set=] of [=strings=].
+1. [=list/For each=] |entry| of |entriesToBatch|:

```

Comment:
Nit: Instead of this, you can do:

`
1. [=Assert=] that:
    `*` |entry|'s ... is equal to ...;
    `*` ...
    `*` |entry|'s ... is equal to ....
`

Reply:
Same as the comment above, this keeps the same format as V1. If we want to change the format, I suggest to create a new PR to change both of them.

Reply:
sure, the follow up PR has several things to do now, so please make sure that nothing will be missed in that PR.